pkg/git/fetch.go

```diff
@@ -25,7 +27,12 @@ func (f Fetcher) Fetch(dir, gitURL, gitRevision, metadataDir string) error {
 		return err
 	}
 
-	repo, err := git.PlainInit(dir, false)
+	tmpDir, err := ioutil.TempDir("", "git-clone-")
+	if err != nil {
+		return err
+	}
+
+	repo, err := git.PlainInit(tmpDir, false)
```

Comment:
Maybe it's worth a comment about we clone to temp directory here. This is something that might become tribal knowledge.

Reply:
I ended up putting it in the commit message, but happy to add a comment if you think we still need one